schema_based_on_diagram.sql

```diff
@@ -0,0 +1,54 @@
+CREATE DATABASE medical_clinic;
+
+CREATE TABLE patients (
+id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+name VARCHAR(100),
+date_of_birth DATE);
+
+CREATE TABLE medical_histories (
+id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+admitted_at DATE,
+patient_id INT,
+status VARCHAR ,
+FOREIGN KEY (patient_id) REFERENCES patients (id));
+
+CREATE INDEX patient_id_index ON medical_histories (patient_id);
+
+CREATE TABLE invoices (
+id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+total_amount DECIMAL ,
+generated_at TIMESTAMP,
+payed_at TIMESTAMP ,
+medical_history_id INT,
+FOREIGN KEY (medical_history_id) REFERENCES medical_histories (id));
+
+CREATE INDEX medical_histories__id_index ON invoices(medical_history_id);
+
+CREATE TABLE invoice_items (
+id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+unit_price DECIMAL,
+quantity INT,
+total_price DECIMAL,
+invoice_id INT,
+treatment_id INT,
+FOREIGN KEY (invoice_id) REFERENCES invoices (id),
+FOREIGN KEY (treatment_id) REFERENCES treatments (id)
+);
+
+
+CREATE INDEX invoice_treatment_id_index ON invoice_items(invoice_id, treatment_id);
+
```

Comment:
- [ ] It would be nice if you make the indexes on both foreign keys in a different query because right now you are making them in composite query which is not the requirement and is not professional in some scenarios for more details kindly read the below details 😊



### Two Separate Indexes (Two Commands):

When you create separate indexes on each foreign key, each index can be used independently by the query planner for optimizing queries that involve those individual foreign keys.
This approach allows for more flexibility in using these indexes in different query scenarios.


### One Composite Index (Single Command):

Creating a single composite index on both foreign keys can be more efficient for certain types of queries that involve both foreign keys together. It can potentially reduce the storage overhead and maintenance costs associated with maintaining multiple indexes.
However, the composite index may not be as efficient for queries that involve only one of the foreign keys, as the composite index might not be used effectively in such cases.